src/core/model/table/typed_column_data.h

```diff
@@ -53,9 +53,14 @@ class TypedColumnData : public model::AbstractColumnData {
 
         MixedType const* mixed = GetIfMixed();
         for (size_t i = 0; i != data_.size(); ++i) {
-            if (GetValueTypeId(i) == +TypeId::kString || GetValueTypeId(i) == +TypeId::kBigInt) {
+            if (GetValueTypeId(i) == +TypeId::kString || GetValueTypeId(i) == +TypeId::kBigInt ||
+                GetValueTypeId(i) == +TypeId::kDate) {
                 std::byte const* value = (mixed) ? mixed->RetrieveValue(data_[i]) : data_[i];
-                StringType::Destruct(value);
+                if (GetValueTypeId(i) == +TypeId::kDate) {
+                    DateType::Destruct(value);
+                } else {
+                    StringType::Destruct(value);
+                }
```

Comment:
Actually, this case-handling doesn't look good, seems like we need a generic method `Type::Destruct<T>` to destruct values which will dispatch to `Destruct` method specific for passed type. 
However, I think, it should be implemented in a separate PR. You may implement it if you want.